homeassistant/components/calendar/caldav.py

```diff
@@ -138,9 +144,11 @@ def update(self):
         # won't return events that have already started
         results = self.calendar.date_search(
             dt.start_of_local_day(),
-            dt.start_of_local_day() + timedelta(days=1)
+            dt.start_of_local_day() + timedelta(days=3)
         )
 
+        results = [x for x in results if "VEVENT" in x.data]
```

Comment:
What happened so far if the item did not contain a "VEVENT"? Does this change current behavior?

Reply:
@tinloaf: Thanks for all the feedback, will get working on it/comment in detail this weekend. The last commit (covering your last two comments) were debugging/bug-fix attempts that I seem to have merged on the same branch and weren't meant to make it into this feature request. Will take those out of the scope of this PR as this needs some more in-depth testing.